java/tests/com/google/privacy/differentialprivacy/ConfidenceIntervalGaussianTest.java

```diff
@@ -0,0 +1,289 @@
+    public abstract static class ConfidenceIntervalTestCase implements Serializable {
+        static ConfidenceIntervalTestCase create(
+                double noisedX,
+                int l0Sensitivity,
+                double lInfSensitivity,
+                double epsilon,
+                Double delta,
+                double alpha,
+                ConfidenceInterval expected) {
+            return new AutoValue_ConfidenceIntervalGaussianTest_ConfidenceIntervalTestCase(
+                    noisedX, l0Sensitivity, lInfSensitivity, epsilon, delta, alpha, expected);
+        }
+
+        abstract double noisedX();
+
+        abstract int l0Sensitivity();
+
+        abstract double lInfSensitivity();
+
+        abstract double epsilon();
+
+        abstract Double delta();
+
+        abstract double alpha();
+
+        abstract ConfidenceInterval expected();
+    }
+
+    @RunWith(Parameterized.class)
+    public static class computeConfidenceIntervalDoubleTests {
+        private final ConfidenceIntervalTestCase testCase;
+
+        public computeConfidenceIntervalDoubleTests(ConfidenceIntervalTestCase testCase) {
+            this.testCase = testCase;
+        }
+
+        @Parameterized.Parameters(name = "{index}: = failed")
+        public static Collection<Object> getTestCases() {
+            return Arrays.asList(
+                    // Random Tests.
+                    ConfidenceIntervalTestCase.create(
+                            70.00,
+                            5,
+                            36,
+                            0.8,
+                            0.8,
+                            0.2,
+                            ConfidenceInterval.create(35.26815080641682, 104.73184919358317)),
+                    ConfidenceIntervalTestCase.create(
+                            699.2402199905,
+                            1,
+                            5,
+                            0.333,
+                            0.9,
+                            0.001256458,
+                            ConfidenceInterval.create(694.5583238637953, 703.9221161172047)),
+                    // High alpha.
+                    ConfidenceIntervalTestCase.create(
+                            70.00,
+                            5,
+                            36,
+                            0.8,
+                            0.8,
+                            1 - 7.856382354e-10,
+                            ConfidenceInterval.create(69.9999999733, 70.0000000267)),
+                    // Low alpha.
+                    ConfidenceIntervalTestCase.create(
+                            70.00,
+                            5,
+                            36,
+                            0.8,
+                            0.8,
+                            7.856382354e-10,
+                            ConfidenceInterval.create(-96.6140883158, 236.6140883158)));
+        }
+
+        @Test
+        public void computeConfidenceIntervalDouble() {
```

Comment:
I don't think this is the proper Java unit test naming. Probably something like `computeConfidenceInterval_hasAccurateResultsForDoubles()` makes more sense.

In general though, you seem to have a class for each test, I thought it is preferable to have a single class for all tests but multiple methods for each test.

But Dennis could correct me on this.